### DISCUSSION TITLE
How to differentiate outside table cell and date table cell?

### DISCUSSION THREAD

**Discussion by BhoomikaSorathiya** [gpbl/react-day-picker#1855] [Support]
### Bug description

In react-day-picker v7, there was `DayPicker-Day--outside` class for outside non-date div. But, for react-day-picker v8, there is no class provided to differentiate between date vs outside date <td/>. 

Basically, I want to provide border to only <td> which has date text. I am trying to do it with `:has()` css selector, but it doesn't support Firefox browser. So looking for alternative like classname with `--outside`. Can anyone please help?

### To reproduce

Go to [here](https://react-day-picker.js.org/basics/selecting-days) for v8 & [here](https://react-day-picker-v7.netlify.app/docs/matching-days) for v7 and inspect element to see class names.

### Expected behavior

There should be som class like `rdp-cell--outside` for outside table cell to differentiate with date cell.

### Screenshots

**react-day-picker v7 `DayPicker-Day--outside` class:**

<img width="1755" alt="Screenshot 2023-07-27 at 6 51 10 PM" src="https://github.com/gpbl/react-day-picker/assets/104916135/6da29c00-1c08-49d4-a4df-cb7a32e6e9e7">

**react-day-picker v8 `rdp-cell` class for outside table cell without any difference from date table cell:**

<img width="1761" alt="Screenshot 2023-07-27 at 6 50 44 PM" src="https://github.com/gpbl/react-day-picker/assets/104916135/af6578f9-d4a5-4b7c-91e3-53c31886b708">

**Comment by gpbl** [comment]
@BhoomikaSorathiya use the `showOutsideDays` prop to show the outside days. There you would see in the DevTool the CSS class is `.rdp-day_outside`. 

<img width="376" alt="Screenshot 2023-07-28 at 05 18 31" src="https://github.com/gpbl/react-day-picker/assets/120693/5b6e966c-43fa-47c7-9316-151d6a5dc341">

> Basically, I want to provide border to only which has date text. I am trying to do it with :has() css selector

You can use 

```css
.rdp-day:not(.rdp-day_outside) {
  border: 1px solid currentColor;
}
```

https://codesandbox.io/s/strange-brown-8jr294?file=/styles.css:533-601

**Reply by BhoomikaSorathiya** [reply]
Thanks @gpbl for quick reply. `showOutsideDays` shows dates which are outside. For my case, I don't want to show `outside` days and it's border.

Currently it's look like this, as I have provided border to all `rdp-cell`. I want to remove border from `rdp-cell` for **outside**.

**Current:**
<img width="366" alt="Screenshot 2023-07-28 at 9 43 56 AM" src="https://github.com/gpbl/react-day-picker/assets/104916135/75d3f581-01a6-4b12-ac74-fdd15dec0e37">


**Expected:**
<img width="365" alt="Screenshot 2023-07-28 at 9 43 48 AM" src="https://github.com/gpbl/react-day-picker/assets/104916135/9fb64029-22c9-4cd2-9326-8cfc83521b79">

**Comment by gpbl** [comment]
@BhoomikaSorathiya thanks for your clarification! (and sorry for the misunderstanding) 

You are right – currently is not possible to get a design like yours, as we are missing an `rd-cell_outside` class here:

https://github.com/gpbl/react-day-picker/blob/main/src/components/Row/Row.tsx#L41-L48

I will work on a fix: https://github.com/gpbl/react-day-picker/issues/1856

**Reply by BhoomikaSorathiya** [reply]
Sounds good. Thanks @gpbl for working on this issue!

**Comment by gpbl** [comment]
The fix for this is included in the [next major version](http://react-day-picker.js.org/next) currently in beta.
